src/Common/MongoProxyClient.ts

```diff
@@ -448,12 +452,13 @@ export async function _createMongoCollectionWithARM(
   }
 
   try {
-    await new ResourceProviderClient(armEndpoint).putAsync(
+    return new ResourceProviderClient<DataModels.CreateCollectionWithRpResponse>(armEndpoint).putAsync(
```

Comment:
@shmotlan This seems like a bug? The upstream code here expects a return value, but it was returning undefined.

Reply:
yes it seems like a bug as its not returning anything. do you still need to keep the `await` before `new ResourceProviderClient`?

Reply:
No. You can omit `await` when returning a promise.

Reply:
ok lgtm